### PR TITLE
Spectrum of MG particles with point source

### PR DESCRIPTION
**Pull request by ChasingNeutrons**
Allow MG particles to have an energy distribution. Also typo fix in fissionSource, bug fix in pointSource, and moved nG from baseMgNeutronDatabase to the superclass

### CODE REVIEW

File: ParticleObjects/Source/pointSource_class.f90 [hunk 1]
```diff
@@ -210,9 +226,16 @@ subroutine sampleEnergy(self, p, rand)
     class(pointSource), intent(inout)   :: self
     class(particleState), intent(inout) :: p
     class(RNG), intent(inout)           :: rand
+    real(defReal)                       :: r
+    integer(shortInt)                   :: g
 
     if (self % isMG) then
-      p % G = self % G
+      r = rand % get()
```
Comment (Mikolaj-A-Kowalski) [inline ParticleObjects/Source/pointSource_class.f90:233]:
```suggestion
      !! TODO: Replace this single, common procedure to sample discrete distributions
      r = rand % get()
```
Sadly the Alias method is implemented... but in the infamous photon branch... :-(

File: NuclearData/mgNeutronData/baseMgNeutron/baseMgNeutronDatabase_class.f90 [hunk 2]
```diff
@@ -45,7 +45,7 @@ module baseMgNeutronDatabase_class
   !!   mats       -> array containing all defined materials (by matIdx)
   !!   majorant   -> majorant xs for delta tracking
   !!   activeMats -> list of matIdxs of materials active in the problem
-  !!   nG         -> number of energy groups
+  !!   nG         -> number of energy groups (from superclass)
```
Comment (Mikolaj-A-Kowalski) [inline NuclearData/mgNeutronData/baseMgNeutron/baseMgNeutronDatabase_class.f90:48]:
```suggestion
```
Maybe we should move this to `mgNeutronDatabase`. (add there `Public Members:` section with nG?

File: NuclearData/mgNeutronData/mgNeutronDatabase_inter.f90 [hunk 3]
```diff
@@ -16,9 +18,10 @@ module mgNeutronDatabase_inter
   !!
   !! It does nothing, It adds nothing,
   !! It just provides a common superclass for related classes
+  !! and holds the number of energy groups
```
Comment (Mikolaj-A-Kowalski) [inline NuclearData/mgNeutronData/mgNeutronDatabase_inter.f90:21]:
```suggestion
  !! and holds the number of energy groups
  !!
  !! Public members: 
  !!   nG -> number of energy groups
```

**Review by Mikolaj-A-Kowalski — APPROVED** [review]
I see no reason not to merge after the few cosmetic changes. 
Thanks for the development!

File: ParticleObjects/Source/pointSource_class.f90 [hunk 4]
```diff
@@ -131,19 +136,30 @@ subroutine init(self, dict, geom)
...
+      nucData => ndReg_getNeutronMG()
+      if(.not.associated(nucData)) call fatalError(Here, 'Failed to retrieve Nuclear Database')
+
+      if (size(temp) /= nucData % nG) then
+        call fatalError(Here, 'Source energy group distribution must have '//numToChar(nucData % nG)//' groups')
+      end if
+
+      allocate(self % probG(nucData % nG))
+      self % probG = temp
+      self % probG = self % probG / self % probG
```
Comment (valeriaRaffuzzi) [inline ParticleObjects/Source/pointSource_class.f90:157]: Should it be divided by the sum?

**Comment by valeriaRaffuzzi** [comment]
What about leaving the option to use only 'G' as well? Sometimes in fixed source one could have lots of groups (vitamin-j is 175), and I've definitely used up to 700. If we want to do fusion, where particles are mono-energetic, it's definitely convenient to leave the option!

File: ParticleObjects/Source/pointSource_class.f90 (hunk 4 above)
Comment (ChasingNeutrons) [reply ParticleObjects/Source/pointSource_class.f90]: Thank you for spotting that. I could swear I did it... Need to write unit tests for sources :/

**Comment by Mikolaj-A-Kowalski** [comment]
Also perhaps we could squash all the commits into one (or few) with `git rebase -i main` (and selecting `squash` [`s`] option for commits that come from the GitHub edits etc...?

**Comment by ChasingNeutrons** [comment]
Ok, after making a slight mess that didn't compile, I think those things should be fixed and now there should be a convenient option for monoenergetic MG particles.

**Comment by ChasingNeutrons** [comment]
Did the squash work as intended?

**Comment by Mikolaj-A-Kowalski** [comment]
Not yet! You may still need to force push your modified branch. (`git push -f`). 
But before doing that it is always good idea to make sure that some backup of the branch being overridden exists somewhere (just in case...) 
Thanks for the rebase! Sorry for being a pain

**Comment by ChasingNeutrons** [comment]
I don't entirely understand what happened but I _think_ that should be it.

**Comment by Mikolaj-A-Kowalski** [comment]
Yup! It is done. Shell I merge?

**Pull request merged**